src/Nethermind/Nethermind.Runner/Program.cs

```diff
@@ -591,4 +557,76 @@ private static string GetProductInfo()
 
         return info.ToString();
     }
+
+    private static bool ValidateArguments(string[] args, CommandLineApplication app)
+    {
+        static HashSet<ReadOnlyMemory<char>> GetValidArguments(CommandLineApplication app)
+        {
+            HashSet<ReadOnlyMemory<char>> validArguments = new(new MemoryContentsComparer<char>());
+            foreach (var option in app.GetOptions())
+            {
+                if (!string.IsNullOrEmpty(option.LongName))
+                {
+                    validArguments.Add(option.LongName.AsMemory());
+                }
+                if (!string.IsNullOrEmpty(option.ShortName))
+                {
+                    validArguments.Add(option.ShortName.AsMemory());
+                }
+            }
+
+            return validArguments;
+        }
+
+        static ReadOnlyMemory<char> GetArgumentName(string arg) => arg.StartsWith("--") ? arg.AsMemory(2) : arg.StartsWith('-') ? arg.AsMemory(1) : ReadOnlyMemory<char>.Empty;
+        static IEnumerable<ReadOnlyMemory<char>> GetArgumentNames(IEnumerable<string> args)
+        {
+            bool lastWasArgument = false;
+            foreach (ReadOnlyMemory<char> potentialArgument in args.Select(GetArgumentName))
+            {
+                if (!lastWasArgument)
+                {
+                    bool isCurrentArgument = lastWasArgument = !potentialArgument.IsEmpty;
+                    if (isCurrentArgument)
+                    {
+                        yield return potentialArgument;
+                    }
+                }
+                else
+                {
+                    lastWasArgument = false;
+                }
+            }
+        }
+
+        static IEnumerable<ReadOnlyMemory<char>> GetDuplicateArguments(IEnumerable<ReadOnlyMemory<char>> argumentsNames)
+        {
+            return argumentsNames.GroupBy(n => n, new MemoryContentsComparer<char>())
+                .Where(g => g.Count() > 1)
+                .Select(g => g.Key);
+        }
+
+        // Get all valid options from the configuration files
+        HashSet<ReadOnlyMemory<char>> validArguments = GetValidArguments(app);
+
+        IEnumerable<ReadOnlyMemory<char>> argumentsNamesProvided = GetArgumentNames(args);
+        foreach (ReadOnlyMemory<char> argumentName in argumentsNamesProvided)
+        {
+            // Check if the argument provided is a valid option/argument
+            if (!validArguments.Contains(argumentName))
+            {
+                throw new ArgumentException($"Unrecognized argument: {argumentName}.\nRun --help for a list of available options and commands.");
```

Comment:
One more thing, maybe better than throwing here would be doing something similar as in duplicate arguments:
1. logging an error.
2. setting ExitCode to specialized (new) exit code
3. returning false.

Reply:
Actually, that's a great suggestion and makes a ton of sense
Let me add that real quick